### DISCUSSION TITLE
인증(로그인) 방식 세션 vs 토큰

### DISCUSSION THREAD

**Discussion by gitchannn** [sunnybraille/sunnybraille-server#111] [General]
어떤 방식으로 인증하는게 우리 서비스에 적절할지에 대한 논의입니다!
사실 아직 사용자가 많지 않아서 어떻게 하든 상관은 없을 것 같아요!
개인적으로는 토큰방식 많이 써보기도 했고, 로그인 유지까지는 데이터베이스를 확실히 덜 거치지만 로그아웃하고 refresh token같은 로직 구현하려면 결국 데이터베이스에 의존적일 수밖에 없어져서 늘 '이럴거면 세션을 그냥 처음부터 할걸'이라고 생각을 해왔었어서 세션을 도입하고 싶기도..?!!
이미 사용해본 방식이 있다면 알려주세요!

[참고](https://engineerinsight.tistory.com/232)

cc. @JunhyeongPark-kr

**Comment by jun-brro** [comment]
찾아보니, 대규모 서비스의 경우 세션 방식을 도입하면 서버 관리가 복잡해진다고 하네요. 다만 현재는 말씀 주셨던 것처럼 사용자가 많지 않아, 세션 방식을 도입해도 괜찮을 듯 합니다.

**Reply by gitchannn** [reply]
넵 그러면 세션으로 일단 개발 시작하겠습니당!

**Reply by gitchannn** [reply]
먼저 간단하게 아이디/비밀번호 통한 로그인/회원가입부터 만들고, 차차 소셜로그인 도입할게

**Comment by gitchannn** [comment]
세션 방식으로 구현하기로 결정해서 close합니다.
